geom/geom/src/TGeoVolume.cxx

```diff
@@ -3028,6 +3028,9 @@ TGeoVolume *TGeoVolumeAssembly::CloneVolume() const
    vol->SetNumber(fNumber);
    vol->SetNtotal(fNtotal);
    vol->SetTitle(GetTitle());
+   // copy extensions
+   vol->SetUserExtension(fUserExtension);
+   vol->SetFWExtension(fFWExtension);
```

Comment:
This will likely trigger `assert(fRC > 0);`  in `TGeoExtension::Release()` when the volumes are destructed, so whenever you give around these pointers you need to call: `TGeoExtension:Grab()` as well to increase the reference counter. So please add:
```
fUserExtension->Grab();
fFWExtension->Grab();
```
And recheck locally (in particular in Debug mode when the assert is compiled)

Reply:
No. Should not be done:
```
void TGeoVolume::SetUserExtension(TGeoExtension *ext)
{
   if (fUserExtension)
      fUserExtension->Release();
   fUserExtension = nullptr;
   if (ext)
      fUserExtension = ext->Grab();
}
```
The initial volume has the refcount up, hence the clone will increase the refcount to 2. 
If I call `fUserExtension->Grab()` explicitly, the refcount goes to 3 and even if both volumes go,
the extension stays  ===> Memory leak.

Reply:
Right, I forgot to check the setter, I thought it was just a simple one. OK then!

Reply:
Actuall, since I see it. The logic is wrong. Should be:
```
void TGeoVolume::SetUserExtension(TGeoExtension *ext)
{
   auto* tmp = fUserExtension;
   fUserExtension = nullptr;
   if (ext)
      fUserExtension = ext->Grab();
   if (tmp)
      tmp->Release();
}
```

Reply:
Please block. I have sent a second comment....